pkg/parse/parse.go

```diff
@@ -601,38 +599,53 @@ func GetBuildOutput(buildOutput string) (define.BuildOutputOption, error) {
 				return define.BuildOutputOption{}, fmt.Errorf("duplicate %q not supported", arr[0])
 			}
 			typeSelected = true
-			if arr[1] == "local" {
-				isDir = true
-			} else if arr[1] == "tar" {
-				isDir = false
-			} else {
+			switch arr[1] {
+			case "local":
+				out.Type = define.BuildOutputLocal
+			case "tar":
+				out.Type = define.BuildOutputTar
+			case "image", "registry":
+				// --type=registry ==> --type=image,push=true
+				out.Type = define.BuildOutputImage
+				out.Image = image
+				imageRef, err := util.ImageStringToImageReference(image)
+				if err != nil {
+					return define.BuildOutputOption{}, fmt.Errorf("failed to convert image to ImageReference")
+				}
+				out.ImageRef = imageRef
+			default:
 				return define.BuildOutputOption{}, fmt.Errorf("invalid type %q selected for build output options %q", arr[1], buildOutput)
 			}
 		case "dest":
 			if pathSelected {
 				return define.BuildOutputOption{}, fmt.Errorf("duplicate %q not supported", arr[0])
 			}
 			pathSelected = true
-			path = arr[1]
+			out.Path = arr[1]
 		default:
 			return define.BuildOutputOption{}, fmt.Errorf("unrecognized key %q in build output option: %q", arr[0], buildOutput)
 		}
 	}
 
-	if !typeSelected || !pathSelected {
-		return define.BuildOutputOption{}, fmt.Errorf("invalid build output option %q, accepted keys are type and dest must be present", buildOutput)
+	if out.Type == define.BuildOutputRegistry {
+		out.Push = true
+	}
+
+	if !typeSelected && !pathSelected {
```

Comment:
This seems to need more nuance. Isn’t the path still required for local destinations, just not for the type `…Image` case?

And, for the type `…Image` case, isn’t specifying a path an error?

Reply:
@danishprakash You never andswered this?

Reply:
I can't recall If I made that change deliberately or if it was a mistake, but you're right, path is required for local dests and not for type.

Reply:
> And, for the type …Image case, isn’t specifying a path an error?

docker seems to not throw an error in this case, and ignores it